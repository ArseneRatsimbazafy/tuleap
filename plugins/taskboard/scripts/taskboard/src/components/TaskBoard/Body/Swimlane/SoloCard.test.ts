/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import SoloCard from "./SoloCard.vue";
import ParentCardRemainingEffort from "./Card/ParentCardRemainingEffort.vue";

describe("SoloCard", () => {
    it("displays the solo card in its own cell", () => {
        const wrapper = shallowMount(SoloCard, {
            propsData: {
                card: {
                    id: 43,
                    remaining_effort: 2.5
                }
            }
        });

        expect(wrapper.contains(SoloCard)).toBe(true);
        expect(wrapper.contains(ParentCardRemainingEffort)).toBe(true);
    });
});
