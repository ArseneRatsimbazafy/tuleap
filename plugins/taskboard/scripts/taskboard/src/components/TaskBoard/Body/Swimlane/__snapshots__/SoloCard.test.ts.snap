// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoloCard displays the parent card in Done column when status maps this column 1`] = `
<div
  class="taskboard-swimlane"
>
  <swimlane-header-stub
    swimlane="[object Object]"
  />
   
  <div
    class="taskboard-cell"
  >
    <!---->
  </div>
  <div
    class="taskboard-cell"
  >
    <solo-card-cell-stub
      card="[object Object]"
    />
  </div>
</div>
`;

exports[`SoloCard displays the parent card in its own cell when status is null 1`] = `
<div
  class="taskboard-swimlane"
>
  <parent-cell-stub
    swimlane="[object Object]"
  />
   
  <div
    class="taskboard-cell"
  />
  <div
    class="taskboard-cell"
  />
</div>
`;

exports[`SoloCard displays the parent card in its own cell when tracker id is not part of columns mapping 1`] = `
<div
  class="taskboard-swimlane"
>
  <parent-cell-stub
    swimlane="[object Object]"
  />
   
  <div
    class="taskboard-cell"
  />
  <div
    class="taskboard-cell"
  />
</div>
`;
